syntax = "proto3";

package bess.pb;

message PCAPPortArg {
  string dev = 1;
}

message PMDPortArg {
  bool loopback = 1;
  oneof port {
    uint64 port_id = 2;
    string pci = 3;
    string vdev = 4;
  }
}

message UnixSocketPortArg {
  string path = 1;
}

message ZeroCopyVPortArg {

}

message VPortArg {
  string ifname = 1;
  oneof cpid {
    string docker = 2;
    int64 container_pid = 3;
    string netns = 4;
  }
  repeated int64 rxq_cpus = 5;
  uint64 tx_tci = 6;
  uint64 tx_outer_tci = 7;
  bool loopback = 8;
  repeated string ip_addrs = 9;
}
